File: publish/day2b.py
#!/usr/bin/python3

def solve(input_fname):
    lines = []
    pos = depth = aim = 0
    with open(input_fname) as ifile:
        for line in ifile:
            lines.append(line.strip().split())
    for cmd, unit in lines:
        unit = int(unit)
        if cmd == "forward":
            pos += unit
            depth += aim * unit
        elif cmd == "up":
            aim -= unit
        elif cmd == "down":
            aim += unit
    return pos * depth


if __name__ == "__main__":
    print(solve("../inputs/day2-demo.in"))

File: cpp/day2b.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string command;
    int value = 0;
    int pos = 0, depth = 0;
    int aim = 0;
    
    while (cin >> command >> value) {
        if (command.compare("up") == 0)
            aim -= value;
        else if (command.compare("down") == 0)
            aim += value;
        else {
            pos += value;
            depth += aim * value;
        }
    }

    cout << pos * depth << endl;

    return 0;
}
